### DISCUSSION TITLE
Job priority setting for production users and individual users

### DISCUSSION THREAD

**Discussion by zhangxiaomei** [DIRACGrid/DIRAC#5909] [Q&A] [answered]
Hi all,

I try to set job priority for the JUNO production group and individual users.   The purpose is to set higher priority for production jobs than user jobs, eg. production jobs: user jobs = 4:1  even more.

According to the guide:

https://dirac.readthedocs.io/en/latest/AdministratorGuide/Systems/WorkloadManagement/JobPriorities/index.html

If Jobsharing is used for the production group, the total priority between user and production group is something like 1:1.  If  we need to set more aggressive priority for production jobs,  do we need to set "UserPriorities" in production jobs?

How is the setting for productions and normal user in other experiments? Thank you!

Best regards,

Xiaomei

**Comment by fstagni** [comment, marked as answer]
Hi Xiaomei, 
the documentation for job sharing and task queues is, I am afraid, incomplete.

You can set a different job shares depending on the user's group that runs the jobs: https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py#L1336

E.g. in LHCb (where we actually give higher shares to users) we have:

```
   lhcb_user  # All LHCb users
   {
      Users = aaa
      ...
      Users += zzz
      Properties = NormalUser
      Properties += PrivateLimitedDelegation
      Quota = 2000
      VOMSRole = /lhcb/Role=user
      JobShare = 10000

    }
    ....
    lhcb_mc  # this is for MonteCarlo simulations
    {
      Users = ...
      Properties = NormalUser
      Properties += JobSharing
      Properties += ProductionManagement
      VOMSRole = /lhcb/Role=production
      JobShare = 300
      AllowBackgroundTQs = True
    } 
    ....
    lhcb_data   # this is for real data productiuons
    {
      Properties = NormalUser
      Properties += JobSharing
      Properties += ProductionManagement
      Properties += LimitedDelegation
      Properties += StageAllowed
      Properties += FileCatalogManagement
      Properties += JobAdministrator
      Properties += Operator
      VOMSRole = /lhcb/Role=production
      JobShare = 40000
      AllowBackgroundTQs = True
    }
```

**Reply by zhangxiaomei** [reply]
Hi Federico,
Thanks, it is clear now.  "JobShare" defines group share,  in LHCb case, user:mc:lhcb = 100:3:400
It is surprised to see in LHCb user group gets much higher share than mc group.

Best regards,
Xiaomei
